spec/services/export/report_spec.rb

```diff
@@ -128,4 +134,12 @@
   def implementing_organisation_column_index
     @headers_for_report.length
   end
+
+  def delivery_partner_organisation_column_index
+    implementing_organisation_column_index + 1
+  end
+
+  def change_state_column_index
+    delivery_partner_organisation_column_index + 1
+  end
```

Comment:
possibly it's just my phobia of the [Connascence of Position](https://connascence.io/position.html), but I fear that this is going to be hard to maintain...

It would be nice to use a helper like:

```rb
  def value_for_attribute(row, attr_name)
    column_name = I18n.t("activerecord.attributes.activity.#{attr_name}")
    row[subject.headers.index(column_name)]
  end
```
but it's not possible as some column names (such as `Delivery partner organisation`) aren't aliases for ActiveRecord attributes (columns in the `activities` table) and aren't listed in any the master list of column names.

Perhaps that's the problem this issue is highlighting? That life could be clearer and simpler if we listed the position of all export fields in one place? e.g. if `Export::ActivityAttributesOrder` became `Export::ActivityColumnsOrder` (and maybe had a separate `Export::ActivityFinancialsColumnsOrder` which we added to the activity metadata field)...

Just an idea, probably there are complexities I'm not seeing! But maybe a candidate for refactoring in due course...

Reply:
Yeah, with hindsight, I would like the data structure to look more like `CSV::Table`, so:

`{"activity-id": { roda_identifier: "value", delivery_partner_identifier: "value", etc}, ...}`

Which would allow better assertions and a lot of other nice things, I think at the time, this felt like a push too far. 😞 

`Export::ActivityAttributesOrder` does define the complete order, including the non Active Record attributes (like `:benefitting_retion` and `:finance`) already.

All really good points though, thank you. There is plenty of potential to refactor here, as usual, I could spend a good year happily improving this code, if only I could! 😆